File: iconics-java/src/main/java/com/open/iconics/context/IconicsContextWrapper.java
package com.open.iconics.context;

import android.content.Context;
import android.content.ContextWrapper;
import android.content.res.Resources;
import android.view.LayoutInflater;

/**
 * https://github.com/chrisjenx/Calligraphy
 */
public class IconicsContextWrapper extends ContextWrapper {

    private LayoutInflater mInflater;

    private IconicsContextWrapper(Context base) {
        super(base);
    }

    public static ContextWrapper wrap(Context base) {
        return new IconicsContextWrapper(base);
    }

    @Override
    public Resources getResources() {
        return super.getResources();
    }

    @Override
    public Object getSystemService(String name) {
        if (LAYOUT_INFLATER_SERVICE.equals(name)) {
            if (mInflater == null) {
                mInflater = new InternalLayoutInflater(LayoutInflater.from(getBaseContext()), this, false);
            }
            return mInflater;
        }
        return super.getSystemService(name);
    }
}

File: iconics-kotlin/src/main/java/com/open/iconics/context/IconicsContextWrapper.kt
package com.open.iconics.context

import android.content.Context
import android.content.ContextWrapper
import android.view.LayoutInflater

/**
 * https://github.com/chrisjenx/Calligraphy
 */
@Deprecated(message = "Use the IconicsImageView or IconicsTextView instead")
class IconicsContextWrapper private constructor(base: Context) : ContextWrapper(base) {

    private val inflater by lazy {
        InternalLayoutInflater(LayoutInflater.from(baseContext), this, false)
    }

    override fun getSystemService(name: String): Any? {
        return if (Context.LAYOUT_INFLATER_SERVICE == name) {
            inflater
        } else {
            super.getSystemService(name)
        }
    }

    companion object {
        @JvmStatic fun wrap(base: Context): ContextWrapper = IconicsContextWrapper(base)
    }
}